### ISSUE TITLE
--includeDeclarations causes typedoc to hang indefinitely

### ISSUE THREAD

**Issue by balupton** [TypeStrong/typedoc#913]
Affecting [eachr#dev-typescript-dts](https://github.com/bevry/eachr/tree/dev-typescript-dts)

Reproduce with:

```
git clone git@github.com:bevry/eachr.git
cd eachr
git checkout dev-typescript-dts
npm install
npm run our:meta:docs:typedoc
```

Hangs with:

```
> eachr@3.2.0 our:meta:docs:typedoc /Users/balupton/Projects/active/eachr
> typedoc --includeDeclarations --mode file --exclude '**/*test*' --name "$npm_package_name" --readme ./README.md --out ./docs ./source


Using TypeScript 3.1.6 from /Users/balupton/Projects/active/eachr/node_modules/typedoc/node_modules/typescript/lib
```

And the version details:

```
> uname -a
Darwin balbook-2018.local 18.2.0 Darwin Kernel Version 18.2.0: Mon Nov 12 20:24:46 PST 2018; root:xnu-4903.231.4~2/RELEASE_X86_64 x86_64

> typedoc --version

TypeDoc 0.13.0
Using TypeScript 3.1.6 from /usr/local/lib/node_modules/typedoc/node_modules/typescript/lib
```

**Comment by aciccarello** [comment]
I was able to run locally in ~190 seconds on a relatively fast computer which is not great but it did complete. I would recommend excluding the node modules directory to avoid unnecessary documentation as that appears to be the main cause of the slowdown. Using the option `--exclude '**/+(*test*|node_modules)/**'` in combination with the changes from #896 brought the build down to under 5 seconds.

**Comment by aciccarello** [comment]
@9oelM What makes you think that TypeDoc isn't hanging on type definition files in `node_modules`?

**Comment by 9oelM** [comment]
@aciccarello woops. did not realize there is a problem in the `exclude` option, especially when excluding `node_modules`, as suggested by many other issues. Hope the #896 is going to be merged soon...

**Comment by Gerrit0** [comment]
With the [v0.14.0](https://github.com/TypeStrong/typedoc/releases/tag/v0.14.0) release, this has been resolved thanks to #890 and #896
